ui/src/shared/components/Crosshair.tsx

```diff
@@ -34,9 +34,17 @@ class Crosshair extends PureComponent<Props> {
   }
 
   private get isVisible() {
-    const {hoverTime} = this.props
+    const {dygraph, hoverTime} = this.props
+    const timeRanges = dygraph.xAxisRange()
+
+    const minTimeRange = timeRanges[0]
+    const isBeforeMinTimeRange = hoverTime < minTimeRange
+
+    const maxTimeRange = timeRanges[1]
+    const isPastMaxTimeRange = hoverTime > maxTimeRange
 
-    return hoverTime !== 0 && _.isFinite(hoverTime)
+    const isValidHoverTime = !isBeforeMinTimeRange && !isPastMaxTimeRange
+    return hoverTime !== 0 && _.isFinite(hoverTime) && isValidHoverTime
```

Comment:
maybe the `isValidHoverTime` as the first or second comparator? i tend to think of checking for validity before whether it meets other criteria

Reply:
that makes sense. i will make that change.